packages/deployer/internal/ast/router-ast-validator.js

```diff
@@ -30,14 +30,61 @@ async function findMissingSelectorsInAST(contracts) {
   return errors;
 }
 
-async function findUnreachableSelectorsInAST() {
-  logger.info('  Unreachable selectors not checked yet in compiled router');
-  return [];
+async function findUnreachableSelectorsInAST(contracts, modules) {
+  const routerSelectors = getCaseSelectors('Router', contracts['Router']);
+  const moduleAddresses = [];
+  for (const [moduleName, moduleData] of Object.entries(modules)) {
+    moduleAddresses[toPrivateConstantCase(moduleName)] = {
+      moduleName,
+      address: moduleData.deployedAddress,
+    };
+  }
+
+  const errors = [];
+  routerSelectors.forEach((s) => {
+    if (
+      !moduleAddresses[s.value.name] ||
+      moduleAddresses[s.value.name].address !== s.value.value.value
+    ) {
+      errors.push({
+        msg: `Selector ${s.selector} not reachable. ${s.value.name} pointing to ${
+          s.value.value.value
+        } instead of ${moduleAddresses[s.value.name]}`,
+      });
+    } else {
```

Comment:
This section will check if the pointed contract has a function with that selector